### PR TITLE
added support for configuring nictype and nicpromise mode in virtualbox

### PR DESCRIPTION
**Pull request by praveen12bnitt**
This has modification required to fix https://github.com/docker/machine/issues/1733

### CODE REVIEW

**Comment by nathanleclaire** [comment]
Cool!

I think it looks pretty good.  `virtio` is also a valid NIC type IIRC, so you might want to add that to the documentation.

@ehazlett WDYT about this?

**Comment by ehazlett** [comment]
+1 -- Yes, add `virtio` and this should be good.  Thanks!

File: docs/drivers/virtualbox.md [hunk 1]
```diff
@@ -29,6 +29,8 @@ Options:
  - `--virtualbox-boot2docker-url`: The URL of the boot2docker image. Defaults to the latest available version.
  - `--virtualbox-import-boot2docker-vm`: The name of a Boot2Docker VM to import.
  - `--virtualbox-hostonly-cidr`: The CIDR of the host only adapter.
+ - `--virtualbox-hostonly-nictype`: Host Only Network Adapter Type. Possible values are are '82540EM' (Intel PRO/1000) and 'Am79C973' (PCnet-FAST III)
```
Comment (nathanleclaire) [inline docs/drivers/virtualbox.md]: To be specific, I think @ehazlett means to add "virtio" to the list of available options here

**Comment by praveen12bnitt** [comment]
done!

**Comment by nathanleclaire** [comment]
LGTM

**Pull request merged**